service/service.go

```diff
@@ -182,41 +132,66 @@ func (app *Application) runAndWaitForShutdownEvent() {
 	}
 }
 
-func (app *Application) shutdownClosableComponents() {
-	for _, closeFn := range app.closeFns {
-		closeFn()
-	}
-}
-
-func (app *Application) setupPipelines() {
-	app.logger.Info("Loading configuration...")
-
+func (app *Application) setupConfigurationComponents() {
 	// Load configuration.
+	app.logger.Info("Loading configuration...")
 	cfg, err := config.Load(app.v, app.factories, app.logger)
 	if err != nil {
 		log.Fatalf("Cannot load configuration: %v", err)
 	}
 
+	app.config = cfg
+
 	app.logger.Info("Applying configuration...")
 
+	app.setupExtensions()
+	app.setupPipelines()
+}
+
+func (app *Application) setupExtensions() {
```

Comment:
Would it be more uniform from code organization perspective to place this function in `builder` package and have an ExtensionsBuilder like we have for other components?

Reply:
Yeah, this will be symmetrical to the others. Writing this one was straightforward and I forgot about the builder. Let's do it later.